server/routes/search.router.js

```diff
@@ -0,0 +1,28 @@
+const express = require('express');
+const pool = require('../modules/pool');
+const router = express.Router();
+const {
+  rejectUnauthenticated,
+} = require('../modules/authentication-middleware');
+
+
+router.get('/:searchTerm', rejectUnauthenticated, (req, res) => {
```

Comment:
This could arguably belong in the `vendors` router. We're searching through vendors, right?

Convention here is either `/api/vendors/search`, or just `/api/vendors?q=whateverSearchTerms`